releasenotes/notes/deprecate-classical-function-3bf44ef26d984366.yaml

```diff
@@ -0,0 +1,11 @@
+---
+deprecations_circuits:
+  - |
+    The :mod:`qiskit.circuit.classicalfunction` and with it
+    the :class:`.ClassicalFunction` and its related :func:`.classical_function`
+    have been deprecated as of Qiskit 1.4 and will be removed in Qiskit 2.0.
+    This change is done to avoid a dependency on the external library `tweedledum`,
+    which is no longer compatible with all of Qiskit's supported platforms and 
+    Python versions.
+    For a similar functionality please use the :class:`.BooleanExpression` which
```

Comment:
but the BooleanExpression class will be part of the circuit library and not of classicalfucntion?

Reply:
actually, will the BooleanExpression class be available? or only the oracles?

Reply:
Currently we only intend to expose the oracles, this release note should be updated to reflect this.